src/os/bluestore/BlueStore.cc

```diff
@@ -3762,6 +3762,12 @@ void *BlueStore::MempoolThread::entry()
 
   bool interval_stats_trim = false;
   while (!stop) {
+    // Update pcm cache settings if osd_memory_target has been changed
+    if (store->osd_memory_target != target && pcm != nullptr) {
```

Comment:
Suggest to use a more generic approach to learn if config has been changed. E.g. some counter incremented on a change and checked against preserved value here...
So you'll universal mechanics for any cache setting update...

Reply:
Done. I have used the counter approach as suggested.